core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java

```diff
@@ -86,7 +88,11 @@ public PrestoSqlDialect(Context context) {
   }
 
   @Override public boolean supportsImplicitTypeCoercion(RexCall call) {
-    return false;
+    return super.supportsImplicitTypeCoercion(call)
+        && RexUtil.isLiteral(call.getOperands().get(0), false)
```

Comment:
this seems to do more than the JIRA issue states: it will not convert a literal except to numeric, date, or timestamp types. But you are not considering the literal type.